smarts/core/mission_planner.py

```diff
@@ -263,9 +264,17 @@ def cut_in_waypoints(self, sim, pose: Pose, vehicle):
             trajectory.append(wp)
         return [trajectory]
 
-    def uturn_waypoints(self, pose: Pose):
+    def uturn_waypoints(self, sim, pose: Pose, vehicle):
         # TODO: 1. Need to revisit the approach to calculate the U-Turn trajectory.
         #       2. Wrap this method in a helper.
+        radius = self._mission.task.trigger_radius
+        neighborhood_vehicles = sim.neighborhood_vehicles_around_vehicle(
+            vehicle=vehicle, radius=radius
+        )
+
+        if not neighborhood_vehicles and not self._trigger_task:
```

Comment:
I do think we should break out a separate `Task.triggered()` method

Reply:
Yes, it's more intuitive to let `Task` maintain the `triggered` status. I tried but turns out there are some drawbacks and limitation:

1. The `Task` becomes mutable. 
2. `Task` object will be used between episodes, which means we have to introduce a mechanism to reset the `Task` when an episode ends.

No.1 is fine, it did raise an `unhashable` error but can be addressed by adding a custom `__hash__` method.
No.2 is more tricky and hard to workaround.

I reverted my changes, just renamed it to `_task_is_triggered`.